frontend/src/common/navbar/index.tsx

```diff
@@ -1,111 +1,82 @@
 import Typography from '@mui/material/Typography';
 import Button from '@mui/material/Button';
 import { useIsLoggedIn } from '@/hooks';
+import { AuthModal } from '@/components/common/AuthModal/AuthModal';
+import { useLogOut } from '@/hooks/useLogOut';
 
-interface NavbarProps {
-  navItems: string[];
-}
+const NavBar = () => {
+  const isLoggedIn = useIsLoggedIn();
+  const [performLogOut, logOutResult] = useLogOut();
 
-interface DrawerProps {
-  navItems: string[];
-  handleDrawerToggle: () => void;
-}
+  const [authModalOpen, setAuthModalOpen] = useState(false); // added this state
 
-const drawerWidth = 240;
-
-const DrawerElement = ({ handleDrawerToggle, navItems }: DrawerProps) => (
-  <Box onClick={handleDrawerToggle} sx={{ textAlign: 'center' }}>
-    <Typography variant="h6" sx={{ my: 2 }}>
-      MUI
-    </Typography>
-    <Divider />
-    <List>
-      {navItems.map((item: string) => (
-        <ListItem key={item} disablePadding>
-          <ListItemButton sx={{ textAlign: 'center' }}>
-            <ListItemText primary={item} />
-          </ListItemButton>
-        </ListItem>
-      ))}
-    </List>
-  </Box>
-);
-
-const NavBar = ({ navItems }: NavbarProps) => {
-  // const window = new Window();
-  const [mobileOpen, setMobileOpen] = React.useState(false);
-  const loggedIn = useIsLoggedIn();
-
-  console.log('logged in? ', loggedIn); //FOR DONGYI -> loggedIn == true -> user is logged in -> show profile, dashboard buttons etc.
-
-  const handleDrawerToggle = () => {
-    setMobileOpen((prevState) => !prevState);
+  const handleLogOut = async (event: React.MouseEvent) => {
+    event.preventDefault(); // prevent default anchor behavior
+    await performLogOut();
+    window.location.href = '/'; // redirect to home after logout
   };
 
-  const container =
-    window !== undefined ? () => window.document.body : undefined;
-
   return (
-    <div style={{ position: 'initial' }}>
-      <AppBar component="nav" position="sticky">
+    <div>
+      <AppBar position="sticky">
         <Toolbar>
-          <IconButton
-            color="inherit"
-            aria-label="open drawer"
-            edge="start"
-            onClick={handleDrawerToggle}
-            sx={{ mr: 2, display: { sm: 'none' } }}
-          >
-            <MenuIcon />
-          </IconButton>
-          <Typography
-            variant="h6"
-            component="div"
-            sx={{ flexGrow: 1, display: { xs: 'none', sm: 'block' } }}
-          >
-            MUI
+          <Typography variant="h6" style={{ flexGrow: 1 }}>
+            KanBan
           </Typography>
-          <Box sx={{ display: { xs: 'none', sm: 'block' } }}>
-            {navItems.map((item) => (
-              <Button key={item} sx={{ color: '#fff' }}>
-                {item}
+
+          {/* If the user is logged in, show dashboard, board, and profile links */}
+          {isLoggedIn && (
+            <>
+              <a href="/" style={{ textDecoration: 'none' }}>
```

Comment:
Also, rather than hardcoding the hrefs, it's better to use the `Routes` enum (in AppRouter from memory).
That way we only define routes in one place, and in the future if we decided to modify them, we'll only have to modify it in one place rather than all over the codebase. Means less work for us :)